tock/utilization/views.py

```diff
@@ -50,3 +57,92 @@ def get_context_data(self, **kwargs):
         context.update({'org_totals': org_billing_context()}
         )
         return context
+
+
+def _plot_utilization(dates, billable, nonbillable):
+    """Make a stacked area plot of billable and nonbillable hours.
+
+    dates, billable, and nonbillable should be sequences with the same length
+    """
+    fig = go.Figure()
+
+    fig.add_trace(go.Scatter(
+        x=dates,
+        y=billable,
+        line_shape="hv",
+        mode="lines+markers",
+        stackgroup="one",
+        name="billable",
+    ))
+    fig.add_trace(go.Scatter(
+        x=dates,
+        y=nonbillable,
+        line_shape="hv",
+        mode="lines+markers",
+        stackgroup="one",
+        name="nonbillable",
+    ))
+
+    fig.update_layout(
+        # autosize=False,
+        # width=900,
+        # height=500,
+        xaxis=dict(autorange=True),
+        yaxis=dict(autorange=True),
+        xaxis_title="Reporting Period Start Date",
+        yaxis_title="Hours",
+        title="Total Hours recorded vs. Time",
+    )
+
+    plot_div = plot(fig, output_type='div', include_plotlyjs=False)
+    return plot_div
+
+
+def _utilization_data(start_date, end_date):
+    Timecard = apps.get_model("hours", "Timecard")
+    billable_filter = Q(timecardobjects__project__accounting_code__billable=True)
+    data = (Timecard.objects.filter(reporting_period__start_date__gte=start_date,
+                                    reporting_period__end_date__lte=end_date,
+        )
+        .values("reporting_period__start_date")
+        .annotate(billable=Sum("timecardobjects__hours_spent", filter=billable_filter))
+        .annotate(nonbillable=Sum("timecardobjects__hours_spent", filter=~billable_filter))
+        .order_by("reporting_period__start_date")
+    )
+    dates = [item["reporting_period__start_date"] for item in data]
+    billable_hours = [item["billable"] for item in data]
+    nonbillable_hours = [item["nonbillable"] for item in data]
+    return dates, billable_hours, nonbillable_hours
+
+
+def utilization_plot(start_date, end_date):
+    """Fetch data and make a plot for total utilization."""
+    return _plot_utilization(*_utilization_data(start_date, end_date))
+
+
+class UtilizationAnalyticsView(PermissionMixin, TemplateView):
+    template_name = 'utilization/utilization_analytics.html'
+
+    def dispatch(self, *args, **kwargs):
+        """
+        Resolve recent reporting periods.
+
+        Although recent_rps is set to the last four reporting periods,
+        we could accept a form response that allows the user or app to
+        dynamically customize number of periods to include in the queryset.
+        """
+        if not self.request.user.is_authenticated:
+            return self.handle_no_permission()
```

Comment:
doctring here looks like a bit of duplication from `GroupUtilizationView` that should be removed.  And in looking at this I think both of these views probably don't need a customized `dispatch` function 😄. I'll file a separate issue to take a closer look at that.